packages/forms/src/UIForm/fieldsets/Array/ArrayItem.component.js

```diff
@@ -31,36 +21,40 @@ export default function ArrayItem(props) {
 					>
 						<Icon name="talend-trash" />
 					</button>
-					{
-						onReorder &&
-						<button
-							disabled={!hasMoveUp}
-							id={id && `${id}-moveUp`}
-							onClick={event => onReorder(event, {
-								previousIndex: index,
-								nextIndex: index - 1,
-							})}
-							title="Move Up"
-							type="button"
-						>
-							<Icon name="talend-caret-down" transform="flip-vertical" />
-						</button>
-					}
-					{
-						onReorder &&
-						<button
-							disabled={!hasMoveDown}
-							id={id && `${id}-moveDown`}
-							onClick={event => onReorder(event, {
-								previousIndex: index,
-								nextIndex: index + 1,
-							})}
-							type="button"
-							title="Move Down"
-						>
-							<Icon name="talend-caret-down" />
-						</button>
-					}
+					{!value.isClosed &&
+						onReorder && (
+							<button
+								disabled={!hasMoveUp}
+								id={id && `${id}-moveUp`}
+								onClick={event =>
+									onReorder(event, {
+										previousIndex: index,
+										nextIndex: index - 1,
+									})
+								}
+								title="Move Up"
+								type="button"
+							>
+								<Icon name="talend-caret-down" transform="flip-vertical" />
+							</button>
+						)}
+					{!value.isClosed &&
+						onReorder && (
+							<button
+								disabled={!hasMoveDown}
+								id={id && `${id}-moveDown`}
+								onClick={event =>
+									onReorder(event, {
+										previousIndex: index,
+										nextIndex: index + 1,
+									})
+								}
+								type="button"
+								title="Move Down"
+							>
+								<Icon name="talend-caret-down" />
+							</button>
+						)}
```

Comment:
That code is duplicated. Please refactor.

Reply:
Not really, those are the 2 reorder arrows. Refactoring it to create a common piece will imply to be able to customize the disabled, id, onClick, title, icon classname.
Basically, all the props except 2 : the `type="button"` and the icon name.

Reply:
Can't you just pass the an `up` or `down` value that generates the proper disabled, id, nextIndex, title and icon?

Reply:
done